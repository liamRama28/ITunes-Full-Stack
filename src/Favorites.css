/* Favorites.css */
.favorites {
    margin-top: 20px;
  }
  
  .favorites h2 {
    font-size: 24px;
    margin-bottom: 10px;
  }
  
  .favorites ul {
    list-style-type: none;
    padding: 0;
  }
  
  .favorites li {
    margin: 10px 0;
  }
  
  .favorites button {
    padding: 5px 10px;
    background-color: #dc3545;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
  }
  
  .favorites button:hover {
    background-color: #c82333;
  }
  