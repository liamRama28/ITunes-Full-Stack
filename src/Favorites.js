import React from 'react';
// Import the API functions you created in api.js
import { searchItunes, addToFavorites } from './api';

const Favorites = ({ favorites, removeFromFavorites }) => {
  return (
    <div>
      <h2>Favorites</h2>
      <ul>
        {favorites.map((favorite) => (
          <div key={favorite.trackId}>
            <li>
              <strong>{favorite.trackName}</strong> - {favorite.artistName}
            </li>
            <button onClick={() => removeFromFavorites(favorite)}>Remove from Favorites</button>
          </div>
        ))}
      </ul>
    </div>
  );
};

export default Favorites;
