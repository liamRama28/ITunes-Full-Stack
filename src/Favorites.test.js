// Favorites.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from './Favorites';

test('Favorites component matches snapshot', () => {
  // Mock favorites data
  const mockFavorites = [
    { trackId: 1, trackName: 'Favorite 1' },
    { trackId: 2, trackName: 'Favorite 2' },
  ];

  // Render the component and create a snapshot
  const component = renderer.create(<Favorites favorites={mockFavorites} />);
  const tree = component.toJSON();

  // Compare the rendered component to the saved snapshot
  expect(tree).toMatchSnapshot();
});
