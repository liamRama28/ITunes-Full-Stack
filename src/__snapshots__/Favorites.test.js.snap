// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component matches snapshot 1`] = `
<div>
  <h2>
    Favorites
  </h2>
  <ul>
    <div>
      <li>
        <strong>
          Favorite 1
        </strong>
         - 
      </li>
      <button
        onClick={[Function]}
      >
        Remove from Favorites
      </button>
    </div>
    <div>
      <li>
        <strong>
          Favorite 2
        </strong>
         - 
      </li>
      <button
        onClick={[Function]}
      >
        Remove from Favorites
      </button>
    </div>
  </ul>
</div>
`;
